src/fsharp/ilx/EraseUnions.fs

```diff
@@ -683,6 +683,32 @@ let convAlternativeDef (addMethodGeneratedAttrs, addPropertyGeneratedAttrs, addP
         | SpecialFSharpOptionHelpers  
         | SpecialFSharpListHelpers  -> 
 
+            let baseTesterMeths, baseTesterProps =
+                if cenv.g.langVersion.SupportsFeature LanguageFeature.UnionIsPropertiesVisible then [], []
```

Comment:
Any idea how to get `langVersion` here?

Reply:
You should be able to pass in a TcGlobals (`g`)

Reply:
Hmm I don't think this is possible because `TcGlobals` is defined later in the file ordering, however adding `langVersion` to `ILGlobals` doesn't seem too invasive.

Reply:
> Hmm I don't think this is possible because TcGlobals is defined later in the file ordering, however adding langVersion to ILGlobals doesn't seem too invasive.

I believe you can move EraseUnions.fs later.  It is really part of IlxGen.fs and can be placed right near there if necessary

Reply:
I took a quick look at this approach, but it seems that `TcGlobals` depends on `EraseUnions`, at least in this one spot: https://github.com/dotnet/fsharp/blob/main/src/fsharp/TcGlobals.fs#L1660.

Perhaps it could be refactored, but I do think the patch to add `langVersion` to `ILGlobals` is fairly tidy (https://github.com/dotnet/fsharp/pull/11394/commits/6885b0b34d4f327b579ca6347257ee2a782176c5), and may come in handy in the future. Any specific objections to that approach?